main.nf

```diff
@@ -238,8 +238,8 @@ process setParalogAnalysis {
     echo "NF internal work directory for [setParalogAnalysis (${task.index})] process:\n\$processDir\n" > $logs_folder/wgd_paralogs.log
 
     if [ \${paranome} = "no" ] && [ \${colinearity} = "no" ]; then
-        echo "[$species] WARNING: Neither whole-paranome analysis nor colinearity analysis is required by configuration file. Exiting."
-        echo "[$species] WARNING: Neither whole-paranome analysis nor colinearity analysis is required by configuration file. Exiting." >> $logs_folder/wgd_paralogs.log
+        echo "[$species] ERROR: Neither whole-paranome analysis nor colinearity analysis has been required in configuration file. Exiting."
+        echo "[$species] ERROR: Neither whole-paranome analysis nor colinearity analysis has been required in configuration file. Exiting." >> $logs_folder/wgd_paralogs.log
```

Comment:
This check seems a bit late in the game here in setParalogAnalysis, shouldn't this be checked much earlier like in checkConfig? And also preferentially in the Python code and not just in the Nextflow pipeline?

What actually happens if both of these are `no`? Will the analysis crash or just do the orthologs?

Reply:
This is definitely checked in the Python code.
But it's true that we could exit much earlier in the NF pipeline, at checkConfig. I'll try that now.
If both are set to "no", the error box print the ERROR sentence of line 242 that is redirected in the log file:
![image](https://user-images.githubusercontent.com/57489957/130267016-f66ee010-9438-419a-a619-4ac5286a6f76.png)

Reply:
> This is definitely checked in the Python code.

But then why the extra check here?

> But it's true that we could exit much earlier in the NF pipeline, at setupAdjustment. I'll try that now.
If both are set to "no", the error box print the ERROR sentence of line 242 that is redirected in the log file:

This seems like it's not checked in the Python code, or at least the check is not executed for some reason when running in Nextflow? I would first check the Python code before redundantly doing it in `setupAdjustment`?

Reply:
Updated. See new commit. The pipeline now stops and shows the ERROR line, which is not anymore redirected to the log file, but it is part of the errorMessage.
![image](https://user-images.githubusercontent.com/57489957/130268855-0936b39f-1fdf-4a83-bed8-6a50078a89d9.png)

Reply:
> What actually happens if both of these are no? Will the analysis crash or just do the orthologs?

I would also still just be curious about the answer to this ;)

Reply:
Well, if they are set both to "no", the pipeline stops with an exit 1 error interrupting everything else including the orthologs...

Reply:
> Updated. See new commit. The pipeline now stops and shows the ERROR line, which is not anymore redirected to the log file, but it is part of the errorMessage.

See my previous comment above. It may seem weird that this check and error is coming from Nextflow while the other checks and errors of the config are coming from the Python code (and go to the log files). Please definitely double-check if this check is also done in the Python code and if so why it doesn't catch this.

Reply:
> Well, if they are set both to "no", the pipeline stops with an exit 1 error interrupting everything else including the orthologs...

Well, yes, I know :) My question is if we would not check for them both to be `no`, would the pipeline then just do the ortholog analyses but no paralog analysis? Or would it crash somehow?

Reply:
> > Updated. See new commit. The pipeline now stops and shows the ERROR line, which is not anymore redirected to the log file, but it is part of the errorMessage.
> 
> See my previous comment above. It may seem weird that this check and error is coming from Nextflow while the other checks and errors of the config are coming from the Python code (and go to the log files). Please definitely double-check if this check is also done in the Python code and if so why it doesn't catch this.

Also, this does seem to be the only such check of the configs within Nextflow that echos an "ERROR: ...".

Reply:
There is a checkpoint in `plot_paralogs.py` (within doRateAdjustment). This is indeed quite late in the pipeline and should be done already in `setup_correction.py` (within setupAdjustment). I'll do this now as a fix, and I'll then also remove the checkpoint made by NF code.

I agree that it would be interesting to not interrupt the pipeline if both are set to "no" and just do all the ortholog-related steps. However the input-output channels in main.nf are now assuming that the paralog/anchor pair distribution is mandatory, and I think that main.nf would need testing and perhaps some edits if we decide to make the paralog steps facultative.

Reply:
> There is a checkpoint in `plot_paralogs.py` (within doRateAdjustment). This is indeed quite late in the pipeline and should be done already in `setup_correction.py` (within setupAdjustment). I'll do this now as a fix, and I'll then also remove the checkpoint made by NF code.

Sounds better!

> I agree that it would be interesting to not interrupt the pipeline if both are set to "no" and just do all the ortholog-related steps. However the input-output channels in main.nf are now assuming that the paralog/anchor pair distribution is mandatory, and I think that main.nf would need testing and perhaps some edits if we decide to make the paralog steps facultative.

OK, at the moment, I was mostly curious what happens in that case. I do not want this to be changed right now. We can keep it in mind and decide about this sometime later.

Reply:
Made the changes about `setup_correction.py` dealing with the checkpoint.